Tasks/google-play-release/GooglePlay.ts

```diff
@@ -102,58 +134,57 @@ function run() {
         // #6) Specify the new change log
         // #7) Commit the edit transaction
 
+        tl.debug(`Getting a package name from ${apkFile}.`);
         let packageName: string = tryGetPackageName(apkFile);
+        updateGlobalParams(globalParams, 'packageName', packageName);
+
+        tl.debug('Initializing JWT.');
         let jwtClient: any = new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
+        globalParams.auth = jwtClient;
+
+        tl.debug('Initializing Google Play publisher API.');
         let edits: any = publisher.edits;
+        [edits, edits.apklistings, edits.apks, edits.tracks, edits.listings, edits.images, jwtClient].forEach(bb.promisifyAll);
 
-        [edits, edits.apklistings, edits.apks, edits.tracks, edits.listings, edits.images, jwtClient].forEach(Promise.promisifyAll);
+        tl.debug('Authorize JWT.');
+        await jwtClient.authorizeAsync();
 
-        globalParams.auth = jwtClient;
-        updateGlobalParams(globalParams, 'packageName', packageName);
+        console.log(tl.loc('GetNewEditAfterAuth'));
+        tl.debug('Creating a new edit transaction in Google Play.');
+        let currentEdit: Edit = await getNewEdit(edits, globalParams, packageName);
+        updateGlobalParams(globalParams, 'editId', currentEdit.id);
 
-        let currentEdit: any = jwtClient.authorizeAsync().then(function (res) {
-            console.log(tl.loc('GetNewEditAfterAuth'));
-            return getNewEdit(edits, globalParams, packageName);
-        });
+        if (shouldUploadApks) {
+            tl.debug(`Uploading ${apkFileList.length} APK(s).`);
 
-        apkFileList.forEach(function (apk) {
-            currentEdit = currentEdit.then(function (res) {
-                console.log(tl.loc('UploadApk', apk));
-                return addApk(edits, apkVersionCodes, packageName, apk, APK_MIME_TYPE);
-            });
-        });
+            for (let apkFile of apkFileList) {
+                tl.debug(`Uploading APK ${apkFile}.`);
```

Comment:
Suggestion: remove the ending period from `tl.debug()` statements that end with filename, path, or version code.